### PR TITLE
[WIP] Handle JavaScript errors in Rust

### PR DESCRIPTION
**Pull request by kitsonk**
This PR will eventually move the handling of JavaScript errors, and remapping of the stack traces from JavaScript to Rust.

Currently it is the first stage, where Rust will log out a string from libdeno of the error from the V8 isolate.

Currently the output looks like this:

```
$ ./target/debug/deno tests/error_001.ts
ERROR RS - Error: bad
/Users/kkelly/github/deno/tests/error_001.ts (foo) 2:11
/Users/kkelly/github/deno/tests/error_001.ts (bar) 5:5
/Users/kkelly/github/deno/tests/error_001.ts () 7:1
gen/bundle/main.js (_gatherDependencies) 133110:16
gen/bundle/main.js (run) 133297:20
gen/bundle/main.js (denoMain) 133569:20
deno_main.js () 1:1
```

### CODE REVIEW

File: libdeno/api.cc [hunk 1]
```diff
@@ -43,8 +43,7 @@ Deno* deno_new(deno_buf snapshot, deno_buf shared, deno_recv_cb cb) {
 }
 
 Deno* deno_new_snapshotter(deno_buf shared, deno_recv_cb cb,
-                           const char* js_filename, const char* js_source,
-                           const char* source_map) {
+                           const char* js_filename, const char* js_source) {
```
Comment (ry) [inline libdeno/api.cc]: This is a welcome change. `source_map` shouldn't have been part of the API, since libdeno did nothing but store it.

**Comment by kitsonk** [comment]
@ry have stage 2, which is the ability for Rust to rewrite the source maps, based on a structure.  The next stage is to plumb it in (and to inline the `main.js.map`)

File: src/msg.fbs [hunk 2]
```diff
@@ -53,6 +53,8 @@ union Any {
   CwdRes,
   Metrics,
   MetricsRes,
+  GetSourceMap,
```
Comment (hayd) [inline src/msg.fbs]: I think you can drop the Get.

File: src/source_maps.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+fn parse_map_string(
+  source_url: &str,
+  get_map: &Fn(&str) -> String,
+) -> Option<Mappings> {
+  parse_mappings::<()>(get_map(source_url).as_bytes()).ok()
+}
+
+fn get_mappings<'a>(
+  source_url: &str,
+  mappings_map: &'a mut HashMap<String, Option<Mappings>>,
```
Comment (hayd) [inline src/source_maps.rs]: It may be worth giving this a type alias (as it is using in several places)

**Comment by ry** [comment]
closed in favor of #1280

**Pull request closed without merging**